cmd/relay-gossip/README.md

```diff
@@ -0,0 +1,235 @@
+
+### Usage from a golang drand client
+
+#### With Group TOML or Chain Info
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alanshaw/drand-gossipsub-client-demo/util"
+	"github.com/drand/drand/client"
+	gclient "github.com/drand/drand/lp2p/client"
+)
+
+const (
+   // listenAddr is the multiaddr the local libp2p node should listen on.
+   listenAddr   = "/ip4/0.0.0.0/tcp/4453"
+   // relayP2PAddr is the p2p multiaddr of the drand gossipsub relay node to connect to.
+   relayP2PAddr = "/ip4/192.168.1.124/tcp/44544/p2p/QmPeerID"
+   // groupTOMLPath is the path to the group configuration information (in TOML format).
+   groupTOMLPath = "/home/user/.drand/groups/drand_group.toml"
+)
+
+func main() {
+	// Create libp2p pubsub
+	ps := util.NewPubsub(listenAddr, relayP2PAddr)
+
+	// Extract chain info from group TOML
+	info := util.ChainInfoFromGroupTOML(groupTOMLPath)
+
+	c, err := client.New(gclient.WithPubsub(ps), client.WithChainInfo(info))
+	if err != nil {
+		panic(err)
+	}
+
+	for res := range c.Watch(context.Background()) {
+		fmt.Printf("round=%v randomness=%v\n", res.Round(), res.Randomness())
+	}
+}
+```
+
+#### With Known Chain Hash
+
+You do not need to know the full group info to use the pubsub client, if you know the chain hash and a HTTP endpoint then you can request the chain info from the HTTP endpoint, verifying it with the known chain hash:
+
+```go
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/alanshaw/drand-gossipsub-client-demo/util"
+	"github.com/drand/drand/client"
+	"github.com/drand/drand/client/http"
+	gclient "github.com/drand/drand/lp2p/client"
+)
+
+const (
+   // listenAddr is the multiaddr the local libp2p node should listen on.
+   listenAddr   = "/ip4/0.0.0.0/tcp/4453"
+   // relayP2PAddr is the p2p multiaddr of the drand gossipsub relay node to connect to.
+   relayP2PAddr = "/ip4/192.168.1.124/tcp/44544/p2p/12D3KooWAe637xuWdRCYkuaZZce13P1F9zJX5gzGUPWZJpsUGUSH"
+   // chainHash is a hash of the group chain information.
+   chainHash    = "c599c267a0dd386606f7d6132da8327d57e1004760897c9dd4fb8495c29942b2"
+   // httpRelayURL is the URL of a drand HTTP API endpoint.
+   httpRelayURL = "http://127.0.0.1:3002"
+)
+
+func main() {
+	// Create libp2p pubsub
+	ps := util.NewPubsub(listenAddr, relayP2PAddr)
+
+	// Chain hash is used to verify endpoints
+	hash, err := hex.DecodeString(chainHash)
+	if err != nil {
+		panic(err)
+	}
+
+	c, err := client.New(
+		gclient.WithPubsub(ps),
+		client.WithChainHash(hash),
+		client.From(http.ForURLs([]string{httpRelayURL}, hash)...),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	for res := range c.Watch(context.Background()) {
+		fmt.Printf("round=%v randomness=%v\n", res.Round(), res.Randomness())
+	}
+}
+```
+
+#### Insecurely
+
+If you trust the HTTP endpoint completely, you don't need chain info or a chain hash:
```

Comment:
HTTP(s) ? it can still come with TLS level of trust, but means you aren't pinning the chain itself in a meaningful way on the client side.